src/desktop/main.cc

```diff
@@ -499,6 +499,11 @@ MAIN {
       } catch (...) {
       }
 
+    #if defined(__APPLE__)
+      if (app.fromSSC) {
```

Comment:
this is only on macOS desktop and when the application is launched from the `ssc` CLI. This allows us to get the exit code of the process, which is impossible from `ssc` as we launch the application with `NSWorkspace` and the actual process that gets launched is **NOT** a child process. So `wait`, `waitpid` and `waitid` will _NOT_ work here :(